### ISSUE TITLE
broken `<ATL>/jest-utils` module resolution in Jest v28

### ISSUE THREAD

**Issue by dzonatan** [testing-library/angular-testing-library#299]
follow-up on https://github.com/facebook/jest/issues/12927

TLDR: `@testing-library/angular/jest-utils` import in Jest v28 is broken and results in this error:
![image](https://user-images.githubusercontent.com/5166666/172627756-bd68ea8c-2183-4c74-bae6-0852e3c488bd.png)

Based on Simen's comment it seems like this library uses a bad package format:
>` @testing-library/angular` needs to add an entry for it in exports.

Which is probably true because other packages like `@angular/core/testing` are following this format. There is even a comment about this in the Angular package format: https://angular.io/guide/angular-package-format#entrypoints-and-code-splitting

Just some comparison images of `@angular/core/testing` which works, and `@testing-library/angular/jest-utils` which doesn't.

`@angular/core/testing`
![image](https://user-images.githubusercontent.com/5166666/172632552-54cbe4ff-7c62-4a6f-8600-1b73f6e0c792.png)

`@testing-library/angular/jest-utils`
![image](https://user-images.githubusercontent.com/5166666/172632449-7e6a84d1-5abf-48c8-84e6-3744eabccd1e.png)
